### ISSUE TITLE
[Bug]: embodied entropy bonus is multiplied by the micro-batch size

### ISSUE THREAD

**Issue by kakiuwang-ui** [RLinf/RLinf#1564]
### Affected component

`rlinf/utils/utils.py::reshape_entropy` as consumed by the embodied actors —
`rlinf/workers/actor/embodied_fsdp_actor_worker.py` and
`rlinf/workers/actor/async_ppo_fsdp_worker.py`.

This one is not metrics-only: the inflated value is multiplied by
`algorithm.entropy_bonus` and subtracted from the loss, so it reaches the gradient.

### Bug description

Both embodied actors aggregate the entropy bonus like this:

```python
entropy = reshape_entropy(
    output_dict["entropy"],
    entropy_type=self.cfg.algorithm.entropy_type,
    action_dim=self.cfg.actor.model.get("action_dim", 7),
    batch_size=output_dict["logprobs"].shape[0],
)
entropy_loss = masked_mean(entropy, mask=loss_mask)
loss -= self.cfg.algorithm.entropy_bonus * entropy_loss
```

`reshape_entropy`'s `chunk_level` branch is `entropy.sum(dim=-1)` — it drops a
dimension. For the models that already reduce entropy to `[bsz, 1]` that leaves a
**rank-1** `[bsz]` tensor, while `loss_mask` is rank-2 `[bsz, 1]` under
`reward_type: chunk_level`. `masked_mean` is

```python
_reduce_sum(values * mask, axis) / _reduce_sum(mask, axis)
```

so `[bsz] * [bsz, 1]` broadcasts into a `[bsz, bsz]` **outer product** in the
numerator, while the denominator still counts the `bsz` entries of the
un-broadcast mask. The result is `sum(entropy)` where `mean(entropy)` was meant —
inflated by exactly the micro-batch size.

The `[bsz, 1]` shape is deliberate on the model side. `openpi_action_model.py:756`:

```python
entropy = entropy.mean(dim=[1, 2, 3], keepdim=False)[
    :, None
]  # [:,None] to align with loss-mask shape
```

`lingbotvla_action_model.py:1151` does the same. `reshape_entropy` then undoes the
alignment that comment is describing.

`cnn_policy` returns `[bsz, action_dim]` and hits the same thing: `sum(dim=-1)`
gives `[bsz]` against the same `[bsz, 1]` mask.

### Configuration YAML file

Any config pairing `entropy_bonus > 0` with `entropy_type: chunk_level`. There are
11 shipped ones; the inflation factor is that config's `actor.micro_batch_size`:

| config | entropy_bonus | micro_batch_size | inflation |
|---|---:|---:|---:|
| `examples/embodiment/config/realworld_peginsertion_async_ppo_cnn.yaml` | 0.005 | 500 | 500x |
| `examples/embodiment/config/realworld_peginsertion_async_ppo_pi05.yaml` | 0.005 | 80 | 80x |
| `examples/embodiment/config/maniskill_ppo_co_training_openpi_pi05.yaml` | 0.005 | 40 | 40x |
| `examples/embodiment/config/maniskill_ppo_openpi.yaml` | 0.005 | 32 | 32x |
| `examples/embodiment/config/maniskill_ppo_openpi_pi05.yaml` | 0.005 | 32 | 32x |
| `examples/embodiment/config/maniskill_async_ppo_openpi.yaml` | 0.005 | 32 | 32x |
| `examples/embodiment/config/maniskill_async_ppo_openpi_pi05.yaml` | 0.005 | 32 | 32x |

plus four `tests/e2e_tests/embodied/` configs whose `micro_batch_size` is 1-4, where
the factor is correspondingly small (1x at `micro_batch_size: 1`).

So `entropy_bonus: 0.005` on `maniskill_ppo_openpi.yaml` acts as `0.16`, and on
`realworld_peginsertion_async_ppo_cnn.yaml` as `2.5`. Changing `micro_batch_size` —
normally a pure memory/throughput knob — silently changes the entropy
regularization along with it.

### Log file

Deterministic CPU reproduction on `main` (`b8c22a39`), no GPU needed:

```python
import torch
from rlinf.utils.utils import masked_mean, reshape_entropy

for mbs in (4, 16, 64, 500):
    entropy = torch.rand(mbs, 1) + 0.5          # openpi / lingbotvla shape
    loss_mask = torch.ones(mbs, 1, dtype=torch.bool)   # reward_type: chunk_level
    e = reshape_entropy(entropy, entropy_type="chunk_level")
    got = masked_mean(e, mask=loss_mask)
    print(f"micro_batch_size={mbs:>4}  entropy{tuple(entropy.shape)} ->"
          f" {tuple(e.shape)}  reported={got.item():>10.4f}"
          f"  true={entropy.mean().item():.4f}"
          f"  ratio={got.item() / entropy.mean().item():>7.2f}x")
```

```text
micro_batch_size=   4  entropy(4, 1) -> (4,)    reported=    3.4850  true=0.8712  ratio=   4.00x
micro_batch_size=  16  entropy(16, 1) -> (16,)  reported=   15.1112  true=0.9444  ratio=  16.00x
micro_batch_size=  64  entropy(64, 1) -> (64,)  reported=   64.8133  true=1.0127  ratio=  64.00x
micro_batch_size= 500  entropy(500, 1) -> (500,) reported=  510.3187 true=1.0206  ratio= 500.00x
```

With only 6 of 16 steps valid the factor is 16.55x rather than a clean 16x, because
the outer product also mixes masked-out rows back in.

### Environment

- RLinf `main` @ `b8c22a39`
- Python 3.13, `torch` CPU
- Shape arithmetic only, so not platform- or backend-specific.

### Other reproduction info

Suggested fix: restore the rank before masking, and broadcast the mask to the
entropy shape so the numerator and denominator count the same elements — the same
treatment `compute_ppo_actor_loss` already gives its own metrics via
`loss_mask.expand_as(ratio)`.

Note that `reshape_entropy` itself should be left alone: the StarVLA action heads
return a 3-D `[bsz, num_action_chunks, action_dim]` entropy, and for those the
`sum(dim=-1)` already lands exactly on the mask's rank. Making the branch
`keepdim=True` instead would fix openpi and break StarVLA.

Two adjacent things I noticed but am not proposing to change here:

1. `reshape_entropy` has no `token_level` branch and no `else`, so an unrecognized
   `entropy_type` silently passes entropy through unchanged. `entropy_type` is not
   validated in `rlinf/config.py` either.
2. `algorithm.loss_agg_func` is documented in `guides/embodiment_config.rst` but the
   embodied actors never read it — they call `masked_mean` directly. Harmless today
   because all 200 embodied configs set `token-mean`, which is that default.

I searched open and closed issues and PRs for `entropy`, `entropy_bonus`,
`reshape_entropy` and `entropy_type` and found nothing covering this. I will open a
PR with the fix and a regression test.

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/resources/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).
